enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.v1.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  SdkIframeEmbedSettings,
+  SdkIframeEmbedTagMessage,
+  SdkIframeEmbedTagSettings,
+} from "./types/embed";
+
+const EMBEDDING_ROUTE = "embed/sdk/v1";
+
+type EmbedSettingKey = keyof SdkIframeEmbedSettings;
+
+const ALLOWED_EMBED_SETTING_KEYS = [
+  "apiKey",
+  "instanceUrl",
+  "dashboardId",
+  "questionId",
+  "template",
+  "theme",
+  "locale",
+] as const satisfies EmbedSettingKey[];
+
+type AllowedEmbedSettingKey = (typeof ALLOWED_EMBED_SETTING_KEYS)[number];
+
+class MetabaseEmbed {
+  static readonly VERSION = "1.0.0";
```

Comment:
When do we change this version? How does this version relate to any of the SDK versions? I guess they're totally different.

Reply:
Yep!

- I think we update this whenever we make (potentially breaking) changes to the script tag, just for the sake of diagnostics. I think we can keep it simple by bumping the major version when we make a breaking change.
- Yes, it'll be totally different than the SDK version. The SDK component is essentially coupled to the Metabase instance's version they're using (e.g. the release-x.54.x branch when the Metabase binary is being built for that release), so it does not correlate with the version on NPM